buildbot_nix/buildbot_nix/repo_config/__init__.py

```diff
@@ -43,10 +45,17 @@ async def extract_during_step(cls, buildstep: BuildStepShellMixin) -> Self:
             )
             return cls()
         try:
-            return cls.model_validate(tomllib.loads(cmd.stdout))
+            config = cls.model_validate(tomllib.loads(cmd.stdout))
+            flake_dir = Path(config.flake_dir).resolve()
+            root_dir = Path.cwd().resolve()
+            flake_dir.relative_to(root_dir)
```

Comment:
We should prevent directory traversal here. The configuration shouldn't allow for ../../flake.nix